python-package/xgboost/sklearn.py

```diff
@@ -442,7 +456,7 @@ def save_model(self, fname: str):
                 meta[k] = v
             except TypeError:
                 warnings.warn(str(k) + ' is not saved in Scikit-Learn meta.')
-        meta['type'] = type(self).__name__
```

Comment:
This is potentially a breaking change. What would happen if the user tried to load an old model from JSON?

Reply:
If they are using xgboost, then it's fine.